lib/internal/fs/promises.js

```diff
@@ -106,8 +106,8 @@ class FileHandle extends JSTransferable {
     return fsync(this);
   }
 
-  read(buffer, offset, length, position) {
-    return read(this, buffer, offset, length, position);
+  read(...args) {
+    return read(this, ...args);
```

Comment:
move this to here maybe better?

```js
  if (arguments.length < 3) {
    // Assume filehandle.read(options)
    ({
      buffer = Buffer.alloc(16384),
      offset = 0,
      length = buffer.length,
      position
    } = buffer);
  }
```

Reply:
Yup, I agree, looks good here, I just did this, waiting for build to finish. Also, I changed ```arguments.length < 3``` to ```arguments.length < 2``` because we don't get filehandle reference argument in this method.